python/seldon_core/seldon_methods.py

```diff
@@ -319,27 +357,43 @@ def transform_output(
             client_response = client_transform_output(
                 user_model, features, datadef.names, meta=meta
             )
-
+            client_response, runtime_metrics, runtime_tags = extract_runtime_data(
+                client_response
+            )
             metrics = client_custom_metrics(
-                user_model, seldon_metrics, OUTPUT_TRANSFORM_METRIC_METHOD_TAG
+                user_model,
+                seldon_metrics,
+                OUTPUT_TRANSFORM_METRIC_METHOD_TAG,
+                runtime_metrics,
             )
 
             return construct_response(
-                user_model, False, request, client_response, meta, metrics
+                user_model,
+                False,
+                request,
+                client_response,
+                meta,
+                metrics,
+                runtime_tags,
             )
         else:
             (features, meta, datadef, data_type) = extract_request_parts_json(request)
             class_names = datadef["names"] if datadef and "names" in datadef else []
             client_response = client_transform_output(
                 user_model, features, class_names, meta=meta
             )
-
+            client_response, runtime_metrics, runtime_tags = extract_runtime_data(
```

Comment:
Looking at what the function is doing, it makes more sense for it to be part of the creation of the client_response above (same for all the ones below)

Reply:
When started working on it inside `predict` method I was initially planning to keep it as `SeldonPrediction` as long as possible but then there was a custom operations on data in `feedback` and `route` methods so maybe your suggestion is better.

Reply:
Movied logic into corresponding methods inside user_model.py file